src/main/java/frc/robot/subsystems/SwerveModule.java

```diff
@@ -46,7 +52,7 @@ public class SwerveModule {
 
   private final ValueCache<Double> m_drivePositionCache;
   private final ValueCache<Double> m_driveVelocityCache;
-  private final ValueCache<Double> m_absoluteRotationCache;
+  private final ValueCache<Angle> m_absoluteRotationCache;
```

Comment:
It looks to me like the encoder API still deals in rotations rather than radians, so this change from `Double` to `Angle`, in combination with the lambda used for the rotation cache, will cause us to be off by a factor of `2*pi`. We can change the lambda (and change the name to `m_absoluteAngleCache`), or we can change this back to `Double` and revert the simplifications in `getAbsoluteRotation2d()`. It looks to me like completing the change over to radians might be best.

Reply:
After further review, your changes look correct. Wow, `Angle` is a brave new world. On the bright side, we don't have to manually track rotations vs radians, but it's probably going to be harder to find individual bugs when we feed the wrong units in somewhere. I have a diff in progress to rename this to `m_absoluteAngleCache`, with no other changes.